cupy_setup_build.py

```diff
@@ -116,7 +116,7 @@
             'amdhip64',  # was hiprtc and hip_hcc before ROCm 3.8.0
             'hipblas',
             'hiprand',
-            'rocfft',
+            'hipfft',
```

Comment:
This line needs to be reverted. `libraries` here indicates the names of the shared libraries that are linked by the builder, so this change is not necessary because there's no `libhipfft.so`, only `librocfft.so`.

Reply:
See
* https://github.com/ROCmSoftwarePlatform/rocFFT/blob/develop/library/include/hipfft.h#L26
* https://github.com/ROCmSoftwarePlatform/hipFFT#transitioning-from-rocfft 

The change is part of ROCm4.1. 

Is 'HIP_VERSION' available in the build scripts already, so we can key off that ?

Reply:
did the actual .so file changed names?

Reply:
@amathews-amd OK. I checked the new hipfft directory structure, and I think the packaging messed up. 

For all other hip* packages, there is a main header e.g. `/opt/rocm-4.1.0/include/hipblas.h` which symlinks to `/opt/rocm-4.1.0/hipblas/include/hipblas.h`. However, at least for Ubuntu's deb `opt/rocm-4.1.0/include` is empty (no symlink to `opt/rocm-4.1.0/hipfft/include/hipfft.h`). The same applies to shared libraries. This is something ROCm/HIP needs to fix, and let's better not make any exception for hipfft since the old way is still working (just a warning at compile time).

Also, this change would break backward compatibility. We've been able to make things work smoothly since ROCm 3.5+ (one of my development machines is still on that!) and I personally disfavor moving as fast as the upstream. Our expected users aren't moving that fast either. It's arguable that everything on HIP is still experimental, but let's do our best here 🙂

Reply:
> Is 'HIP_VERSION' available in the build scripts already, so we can key off that ?

Sorry, forgot to reply on this. Yes, we do set `HIP_VERSION` at the build time, but not here but later, which is why I said it'd break the backward compatibility.

Reply:
@emcastillo, yes

@leofang 
Right now, things should still work the old; Just print a 'deprecated' message. I will reverse the changes in the build scripts for now.

Reply:
> did the actual .so file changed names?

@emcastillo Sorry, for some reason GitHub didn't show your reply promptly. 

Since ROCm 4.1.0 there is a new `opt/rocm-4.1.0/lib/libhipfft.so` symlinking to `opt/rocm-4.1.0/hipfft/lib/libhipfft.so`.

Reply:
> > did the actual .so file changed names?
> 
> @emcastillo Sorry, for some reason GitHub didn't show your reply promptly.
> 
> Since ROCm 4.1.0 there is a new `opt/rocm-4.1.0/lib/libhipfft.so` symlinking to `opt/rocm-4.1.0/hipfft/lib/libhipfft.so`.

OK so for this particular line it's ok to change for ROCm 4.1.0.

Reply:
Had a quick chat offline with @kmaehashi, let's keep this so name change here as the team is interested in seeing the support for ROCm 4.1.0 to land. Backward compatibility starts with 4.1.0.

But, @amathews-amd it'd be great if you could kindly ask your team to fix the directory structure (https://github.com/cupy/cupy/pull/5067#discussion_r610286064) 🙂

Reply:
I discussed the directory structure issue internally with Emmett. 
New issue opened on rocFFT to track this : https://github.com/ROCmSoftwarePlatform/rocFFT/issues/312

Reply:
Thanks @amathews-amd!